packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx

```diff
@@ -126,6 +126,31 @@ describe('<BaseButton />', () => {
     expect(onClick).toHaveBeenCalledTimes(1);
   });
 
+  it('should call function on click of button with event callback', () => {
+    const buttonText = 'Pay Now';
+    const onClick = jest.fn();
+    const { getByRole } = renderWithTheme(<BaseButton onClick={onClick}>{buttonText}</BaseButton>);
+    const button = getByRole('button');
+    fireEvent(
+      button,
+      new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true,
+        screenX: 20,
+        screenY: 20,
+      }),
+    );
```

Comment:
import {userEvent} from 'react-testing-library/events'
const user = userEvent.setup()

user.click()

Reply:
Need to assert if callback has been called. Couldn't find a way to add that assertion with userEvents. Have you added such an assertion before? If not we can just keep it this way as long as our test is able to assert on the callback